jobqueue/scheduler/openstack.go

```diff
@@ -1072,10 +1077,8 @@ func (s *opst) cmdNotNeeded(cmd string) {
 // if we can't start the cmd, not if the command fails (schedule() only
 // guarantees that the cmds are run count times, not that they are /successful/
 // that many times).
-func (s *opst) runCmd(cmd string, req *Requirements, reservedCh chan bool, call string) error {
-	logger := s.Logger.New("call", call)
-
-	requestedOS, requestedScript, requestedConfigFiles, requestedFlavor, needsSharedDisk, err := s.serverReqs(req)
+func (s *opst) runCmd(ctx context.Context, cmd string, req *Requirements, reservedCh chan bool, call string) error {
+	requestedOS, requestedScript, requestedConfigFiles, requestedFlavor, needsSharedDisk, err := s.serverReqs(ctx, req)
```

Comment:
Instead of taking a call value, the caller of runCmd should set call value in the ctx.